raft/tracker/progress.go

```diff
@@ -235,3 +236,29 @@ func (pr *Progress) String() string {
 	}
 	return buf.String()
 }
+
+// A ProgressMap is a map of *Progress.
+type ProgressMap map[uint64]*Progress
+
+// String prints the ProgressMap in sorted key order, one Progress per line.
+func (m ProgressMap) String() string {
+	type tup struct {
```

Comment:
nit: I think the idiomatic way to do this is:
```
ids := make([]uint64, 0, len(m))
for k := range m {
    ids = append(ids, k)
}
sort.Slice(tups, func(i, j int) bool {
    return ids[i] < ids[j]
})
var buf strings.Builder
for _, id := range ids {
    fmt.Fprintf(&buf, "%d: %s\n", id, m[id])
}
return buf.String()
```
It's cleaner and requires fewer lines of code.